server/controllers/static.ts

```diff
@@ -95,6 +103,45 @@ async function getVideoCaption (req: express.Request, res: express.Response) {
   return res.sendFile(path, { maxAge: STATIC_MAX_AGE })
 }
 
+async function generateNodeinfo (req: express.Request, res: express.Response, next: express.NextFunction) {
+  const { totalVideos: totalVideos } = await VideoModel.getStats()
+  const { totalLocalVideoComments: totalLocalVideoComments } = await VideoCommentModel.getStats()
+  const { totalUsers } = await UserModel.getStats()
+
+  const json: HttpNodeinfoDiasporaSoftwareNsSchema20 = {
+    version: '2.0',
+    software: {
+      name: 'PeerTube',
+      version: '1.0.0-beta.9'
```

Comment:
Fetch the version from package.json (see config.ts controller)

Reply:
Oh. It's so simple it hurts :')